### PR TITLE
GH-32: Rework `main` for poetry plugin

### PR DESCRIPTION
**Pull request by eugenetriguba**
This reworks the CLI entry point to separate them out into a `main` function and a `run` function. This `run` function takes in a list of arguments, an optional working directory to use, and returns the exit code.

This restructure is necessary for the implementation of the poetry plugin. Since the "task" command is being added as a sub command to the poetry CLI, it needs to integrate with it's CLI framework. By making how this CLI is run a little more flexible, it makes that integration much easier to do.

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -1,7 +1,6 @@
 [MASTER]
```
Comment (eugenetriguba) [inline .pylintrc:1]: note: With the newer pylint version:

```
************* Module /home/eugene/src/taskipy/.pylintrc
.pylintrc:1:0: E0015: Unrecognized option found: missing-module-docstring (unrecognized-option)
.pylintrc:1:0: R0022: Useless option value for '--disable', 'R0201' was moved to an optional extension, see https://pylint.readthedocs.io/en/latest/whatsnew/2/2.14/summary.html#removed-checkers. (useless-option-value)
```

File: pyproject.toml [hunk 2]
```diff
@@ -25,15 +25,27 @@ tomli = [
     { version = "^2.0.1", python = "^3.7" },
     { version = "^1.2.3", python = "~3.6" }
 ]
-psutil = "^5.7.2"
+psutil = "^5.9.4"
 mslex = { version = "^0.3.0", markers = "sys_platform == 'win32'" }
-colorama = "^0.4.4"
+colorama = [
+    { version = "^0.4.6", python = "^3.7" },
```
Comment (eugenetriguba) [inline pyproject.toml:31]: note: 0.4.6 drops python 3.6
- https://github.com/tartley/colorama/blob/master/CHANGELOG.rst

File: tests/test_taskipy.py [hunk 3]
```diff
@@ -26,12 +26,23 @@ def tearDown(self):
         for tmp_dir in self._tmp_dirs:
             tmp_dir.clean()
 
-    def run_task(self, task: str, args: List[str] = None, cwd=os.curdir) -> Tuple[int, str, str]:
+    def run_task(
+        self,
+        task: str,
+        args: Optional[List[str]] = None,
```
Comment (eugenetriguba) [inline tests/test_taskipy.py:32]: note: mypy linting changes. The type hint is saying `args` is a list of strings but then is setting it to `None`.

**Comment by eugenetriguba** [comment]
note: I had, as mentioned in Discord, looked into running updated dependencies for py37+, but I ran into some issues with the python 3.6 CI test run. I've reverted it from here and will look into it elsewhere.

**Comment by illBeRoy** [comment]
overall looks great! can you please elaborate a little bit about why we need to separate `main` from `run`? thanks!

**Comment by eugenetriguba** [comment]
@illBeRoy It turns out I needed an abstraction over the IO as well so there is a very small change in regard to that, will resolve the failed tests in a moment.

The gist is that this needs to integrate with Poetry's CLI framework (cleo). It's an opinionated framework that has it's own way of handling IO, arguments, options, etc. By separating out and making things a bit more customizable and flexible, it's much easier to integrate with it.

**Comment by eugenetriguba** [comment]
There was a similar approach taken from this closed PR: https://github.com/taskipy/taskipy/pull/33/files#diff-f57785de79a15e29053d5ebf8262613cefb2a76ed9c77e1775076afb6c505650R29

**Comment by eugenetriguba** [comment]
So I'm not sure we're going to have a fantastic way to integrate nicely with their IO system. Even with something like this (https://github.com/taskipy/taskipy/pull/60/commits/added63052bae85115f32853f735821ca44ba674) and passing in an IO object that utilizes poetry's IO system, that doesn't take into account that the vast majority of the IO here is through the tasks being run. I wouldn't want to touch the output of those tasks. It sounds bound to be error prone to me. I'm going to leave it out

**Review by illBeRoy — APPROVED** [review]

**Pull request merged**